android/src/main/java/io/keen/client/android/AndroidJsonHandler.java

```diff
@@ -40,8 +44,27 @@
 
         String json = readerToString(reader);
         try {
-            JSONObject jsonObject = new JSONObject(json);
-            return JsonHelper.toMap(jsonObject);
+            Object jsonObjOrArray = getJsonObjectManager().newTokener(json).nextValue();
+
+            // TODO : What should the return value be? Technically it could be a List or Map, so it
+            // should be Object, then client code would need to do the instanceof check. For now,
+            // so as to not break the KeenJsonHandler interface, we can stick a dummy "root" key in
+            // the map we pass back.
```

Comment:
The problem is that the JSON parser will return either a Map<> or a List<> depending on the root JSON structure. Usually the root is a JSON Object, but now we have to handle a JSON Array as a root object too. So in the `vnext_major` branch we should choose whether to branch here or at a higher layer of abstraction. This TODO is mostly an inquiry to you folks in case you have an opinion as to where that branching should occur, but it doesn't have to be decided today...and if you have no opinion then I'll just do what I think is best in the `vnext_major` branch and leave this as is for the minor release.